### PR TITLE
implemented ckcp CI with pipeline-service

### PR DESCRIPTION
**Pull request by xinredhat**
This PR is to implement ckcp CI with Pipelinerun CR, which can be triggered on pipeline-serivce when PR or Push occurs. The exiting CI will be removed from openshift CI https://github.com/openshift/release/pull/33909

### CODE REVIEW

File: .tekton/pipeline-service-ckcp-testing.yaml [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+
+                kubectl -n default exec pod/podman -- sh -c "yum install -y python3-pip xz wget git make jq gcc && yum clean all"
+                kubectl -n default exec pod/podman -- sh -c "wget -c https://go.dev/dl/go1.19.1.linux-amd64.tar.gz && tar -C /usr/local -xzf go1.19.1.linux-amd64.tar.gz && rm -rf go1.19.1.linux-amd64.tar.gz"
+                kubectl -n default exec pod/podman -- sh -c "curl --location -o /usr/local/bin/argocd https://github.com/argoproj/argo-cd/releases/download/v2.4.7/argocd-linux-amd64 && chmod +x /usr/local/bin/argocd && argocd version --client"
+                kubectl -n default exec pod/podman -- sh -c "KUBECTL_VERSION=1.24.3 && curl -LO https://storage.googleapis.com/kubernetes-release/release/v\$KUBECTL_VERSION/bin/linux/amd64/kubectl && chmod +x ./kubectl && mv ./kubectl /usr/local/bin && kubectl version --client=true"
+                kubectl -n default exec pod/podman -- sh -c "wget https://github.com/mikefarah/yq/releases/download/v4.25.1/yq_linux_amd64 -O /usr/local/bin/yq && chmod +x /usr/local/bin/yq && yq --version"
+            - name: ckcp-setup
+              image: quay.io/xinjiang/kubectl:1.25
```
Comment (bnallapeta) [inline .tekton/pipeline-service-ckcp-testing.yaml:198]: Change this to the official image. If this is a custom image you've built, then please upload this to quay.io/redhat-pipeline-service organization.

File: .tekton/pipeline-service-ckcp-testing.yaml [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+          params:
+            - name: deploy-cluster-status
+            - name: namespace
+            - name: cluster-name
+          workspaces:
+            - name: kubeconfig-dir
+          kind: ClusterTask
+          steps:
+            - name: destroy
+              image: quay.io/xinjiang/kubectl:1.25
```
Comment (bnallapeta) [inline .tekton/pipeline-service-ckcp-testing.yaml:261]: Change this to the official image. If this is a custom image you've built, then please upload this to quay.io/redhat-pipeline-service organization.

File: .tekton/pipeline-service-ckcp-testing.yaml [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+                      readOnly: true
+                  volumes:
+                  - name: pull-secret
+                    secret:
+                      secretName: pull-secret
+                EOF
+                kubectl -n default wait pod/podman --for=condition=Ready --timeout=90s
+                kubectl -n default describe pod/podman
+            - name: install-softwares-to-podman-container
+              image: quay.io/xinjiang/kubectl:1.25
```
Comment (bnallapeta) [inline .tekton/pipeline-service-ckcp-testing.yaml:183]: Change this to the official image. If this is a custom image you've built, then please upload this to quay.io/redhat-pipeline-service organization.

File: .tekton/pipeline-service-ckcp-testing.yaml [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+          - name: kubeconfig-dir
+            workspace: shared-workspace
+        taskSpec:
+          workspaces:
+            - name: source
+            - name: kubeconfig-dir
+          kind: ClusterTask
+          steps:
+            - name: create-podman-container
+              image: quay.io/xinjiang/kubectl:1.25
```
Comment (bnallapeta) [inline .tekton/pipeline-service-ckcp-testing.yaml:143]: Change this to the official image. If this is a custom image you've built, then please upload this to quay.io/redhat-pipeline-service organization.

File: .tekton/pipeline-service-ckcp-testing.yaml [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+                  value: "$(params.image)"
+              script: |
+                envsubst < $(workspaces.hypershift-cluster-template.path)/cluster-template.yaml | tee $(workspaces.output.path)/HypershiftDeployment.yaml
+            - name: deploy-cluster
+              image: quay.io/xinjiang/kubectl:1.25
+              script: |
+                #!/usr/bin/env bash
+                set -o errexit
+                export KUBECONFIG=$(workspaces.kubeconfig-dir.path)/kubeconfig
+                kubectl -n $(params.namespace) apply -f $(workspaces.output.path)/HypershiftDeployment.yaml
```
Comment (bnallapeta) [inline .tekton/pipeline-service-ckcp-testing.yaml:97]: Best practice is to name a file with lower case with underscores between words.
hypershift_deployment.yaml

File: .tekton/pipeline-service-ckcp-testing.yaml [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+                - name: REGION
+                  value: "$(params.region)"
+                - name: CLUSTER_NAME
+                  value: "$(params.cluster-name)"
+                - name: IMAGE
+                  value: "$(params.image)"
+              script: |
+                envsubst < $(workspaces.hypershift-cluster-template.path)/cluster-template.yaml | tee $(workspaces.output.path)/HypershiftDeployment.yaml
+            - name: deploy-cluster
+              image: quay.io/xinjiang/kubectl:1.25
```
Comment (bnallapeta) [inline .tekton/pipeline-service-ckcp-testing.yaml:92]: Change this to the official image. If this is a custom image you've built, then please upload this to quay.io/redhat-pipeline-service organization.

File: .tekton/pipeline-service-ckcp-testing.yaml (hunk 2 above)
Comment (xinredhat) [reply .tekton/pipeline-service-ckcp-testing.yaml:261]: ah, thanks for pointing this issue. I forgot it

File: .tekton/pipeline-service-ckcp-testing.yaml (hunk 1 above)
Comment (xinredhat) [reply .tekton/pipeline-service-ckcp-testing.yaml]: fixed. Copied the image docker.io/bitnami/kubectl:1.25 to quay.io/redhat-pipeline-service organization.

File: .tekton/pipeline-service-ckcp-testing.yaml (hunk 2 above)
Comment (xinredhat) [reply .tekton/pipeline-service-ckcp-testing.yaml]: fixed, same as above

File: .tekton/pipeline-service-ckcp-testing.yaml (hunk 3 above)
Comment (xinredhat) [reply .tekton/pipeline-service-ckcp-testing.yaml]: fixed, same as above

File: .tekton/pipeline-service-ckcp-testing.yaml (hunk 4 above)
Comment (xinredhat) [reply .tekton/pipeline-service-ckcp-testing.yaml]: fixed, same as above

File: .tekton/pipeline-service-ckcp-testing.yaml (hunk 5 above)
Comment (xinredhat) [reply .tekton/pipeline-service-ckcp-testing.yaml]: fixed

File: .tekton/pipeline-service-ckcp-testing.yaml (hunk 6 above)
Comment (xinredhat) [reply .tekton/pipeline-service-ckcp-testing.yaml]: fixed, same as above

**Review by bnallapeta — APPROVED** [review]
/lgtm

**Pull request merged**